mason external failed test

ankingcodes opened the pull request:
Due to latest changes to spack, the `libtomlc99` test failed. 
As suggested by spack maintainers, we fetch `releases/latest` instead of `master`. However, due to unexpected complications on `latest`, `libtomlc99` still fails. 
```
$  mason search libtomlc99
==> 1 packages.
libtomlc99

$ mason external install libtomlc99
==> Error: name 'XorgPackage;' is not defined
Package could not be installed
```

As a result, adding a `.notest` to this test so that nightly testing passes as we investigate the issue further.

bradcray commented:
I'm curious what impact the spack change will have on the released 1.22.0 version of mason.  Did we use the master branch then, or was that before that change went into the code base?  And does that suggest that we should make the branch of spack to use configurable to avoid getting stuck with a bad branch name in the future?

ankingcodes commented:
@bradcray I believe we've always used master. When I made the change , I had split up spack-registry to its "CLI" and "registry" 
 and updated the version of the tag in my PR before 1.22 release. See the PR here : https://github.com/chapel-lang/chapel/pull/15641. 
> And does that suggest that we should make the branch of spack to use configurable to avoid getting stuck with a bad branch name in the future?

I believe we should take some precautions so that changes to spack doesn't break Mason.
